apps/balancer-tools/src/app/poolsimulator/(components)/SwapSimulator.tsx

```diff
@@ -290,3 +143,293 @@ function calculateSimulation({
+                    {swapType}
+                  </SelectItem>
+                ))}
+              </Select>
+            )}
+          />
+        </div>
+        <div>
+          <FormField
+            name="amount"
+            render={({ field }) => (
+              <Input
+                {...field}
+                label="Amount"
+                type="number"
+                defaultValue={defaultAmount}
+              />
+            )}
+          />
+        </div>
+        <div className="col-span-2 mt-4">
+          <Button className="w-full" type="submit">
+            Re-calculate swap
+          </Button>
+        </div>
+      </div>
+    </Form>
+  );
+}
+
+function ErrorMessage({
+  errorMessage,
+  errorTitle,
+}: {
+  errorMessage: string;
+  errorTitle: string;
+}) {
+  return (
+    <div role="alert">
+      <div className="bg-tomato9 text-slate12 font-bold rounded-t px-4 py-2">
+        Error: {errorTitle}
+      </div>
+      <div className="border border-t-0 border-red-400 rounded-b bg-tomato12 px-4 py-3 text-tomato7">
+        <p>{errorMessage}</p>
+      </div>
+    </div>
+  );
+}
+
+function SimulationResult({
+  swapType,
+  tokenInSymbol,
+  tokenOutSymbol,
+  data,
+  amountIn,
+  amountOut,
+  effectivePrice,
+  priceImpact,
+}: {
+  swapType: string;
+  tokenInSymbol: string;
+  tokenOutSymbol: string;
+  data: AnalysisData;
+  amountIn?: number;
+  amountOut?: number;
+  effectivePrice?: number;
+  priceImpact?: number;
+}) {
+  if (!data.poolType) {
+    return <Spinner />;
+  }
+  // When CLP is out of bounds in an exactIn swap it returns 0
+  if (
+    (data.poolType === PoolTypeEnum.Gyro2 ||
+      data.poolType === PoolTypeEnum.Gyro3 ||
+      data.poolType === PoolTypeEnum.GyroE) &&
+    (!amountOut ||
+      amountOut >
+        (data.tokens.find((t) => t.symbol === tokenOutSymbol)
+          ?.balance as number))
+  ) {
+    return (
+      <ErrorMessage
+        errorTitle="CLP limit"
+        errorMessage="The swap is greater than the pool limit. Please, change the amount to a lower value."
+      />
+    );
+  }
+  if (!amountIn || !amountOut || !effectivePrice || !priceImpact)
+    return <Spinner />;
+
+  return (
+    <div className="flex flex-col gap-y-2">
+      {swapType == "Exact In" ? (
+        <>
+          <Label className="block text-sm text-slate12">Amount Out</Label>
+          <BaseInput
+            label="Amount Out"
+            value={`${amountOut.toFixed(2)} ${tokenOutSymbol}`}
+            disabled
+          />
+        </>
+      ) : (
+        <>
+          <Label className="block text-sm text-slate12">Amount In</Label>
+          <BaseInput
+            value={`${amountIn.toFixed(2)} ${tokenInSymbol}`}
+            disabled
+          />
+        </>
+      )}
+      <Label className="block text-sm text-slate12">Effective Price</Label>
+      <BaseInput
+        value={`${effectivePrice.toFixed(
+          2,
+        )} ${tokenInSymbol}/${tokenOutSymbol}`}
```

Comment:
👍 I think we might use this in other places, and would reuse the same function we'd created to parse this if that's the case